ttnn/cpp/ttnn/kernel/kernel_host_utils.hpp

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+#if !defined(KERNEL_BUILD)
+
+#include <type_traits>
+#include <vector>
+#include <bit>
+
+namespace ttnn::kernel_utils {
+
+template <typename KernelStruct>
+concept KernelArgsStructU32Concept =
```

Comment:
May I suggest a name that clarifies intent a bit better like `SerializableKernelArgs`? It's a bit odd to include the word `Concept` in the name itself. Also `alignof(KernelStruct) == alignof(uint32_t)` implies `(sizeof(KernelStruct) % sizeof(uint32_t)) == 0` so you don't need to include both of those, just the alignment check.